src/bin/cargo/main.rs

```diff
@@ -277,6 +304,7 @@ fn search_directories(config: &Config) -> Vec<PathBuf> {
 }
 
 /// Initialize libgit2.
+#[tracing::instrument(skip_all)]
 fn init_git(config: &Config) {
```

Comment:
Aside: @Byron do you know if there should be or will be an equivalent to this in gitoxide?  In the PR description is a trace that shows that this operation is 20% of the no-op `cargo check` time.  Wondering if this is something that is worth hacking around or if its a "this will go away with git2".

Reply:
This is an interesting case as its unclear to me how much 20% actually represent in absolute terms of time.
With that said, I looked at what `init_git` is doing and I was surprised it takes time at all - it *felt* like it's loading some code dynamically (which is probably wrong).

`gitoxide` isn't comparable with `git2` as there are no globals, and everything is lazy, so I'd guess that the timing is going to be more favourable. For one, there is no equivalent to `init_git` - configuration is done only when opening the repository.

I hope that helps.